src/openvpn/buffer.c

```diff
@@ -438,11 +438,12 @@ format_hex_ex (const uint8_t *data, int size, int maxoutput,
 	       unsigned int space_break_flags, const char* separator,
 	       struct gc_arena *gc)
 {
-  struct buffer out = alloc_buf_gc (maxoutput ? maxoutput :
-				    ((size * 2) + (size / (space_break_flags & FHE_SPACE_BREAK_MASK)) * (int) strlen (separator) + 2),
-				    gc);
   int i;
+  struct buffer out;
   ASSERT( maxoutput > 0 || separator != NULL );
+  out = alloc_buf_gc (maxoutput ? maxoutput :
+				    ((size * 2) + (size / (space_break_flags & FHE_SPACE_BREAK_MASK)) * (int) strlen (separator) + 2),
+				    gc);
   for (i = 0; i < size; ++i)
     {
       if (separator && i && !(i % (space_break_flags & FHE_SPACE_BREAK_MASK)))
```

Comment:
Hi,

On Wed, Jun 08, 2016 at 10:02:03AM -0700, Selva Nair wrote:
> >    ASSERT( maxoutput > 0 || separator != NULL );
> > +  out = alloc_buf_gc (maxoutput ? maxoutput :
> > +				    ((size * 2) + (size / (space_break_flags & FHE_SPACE_BREAK_MASK)) * (int) strlen (separator) + 2),
> > +				    gc);
> >    for (i = 0; i < size; ++i)
> >      {
> >        if (separator && i && !(i % (space_break_flags & FHE_SPACE_BREAK_MASK)))
> 
> To nitpick, with the ASSERT added, now this additional check on 'separator'  is no more needed.

It is.  We permit separator being NULL *if* maxoutput is > 0 ...

gert
-- 
USENET is *not* the non-clickable part of WWW!
                                                           //www.muc.de/~gert/
Gert Doering - Munich, Germany                             gert@greenie.muc.de
fax: +49-89-35655025                        gert@net.informatik.tu-muenchen.de